alica_engine/src/engine/AlicaEngine.cpp

```diff
@@ -35,6 +36,7 @@ AlicaEngine::AlicaEngine(AlicaContext& ctx, const std::string& configPath,
                          const std::string& roleSetName, const std::string& masterPlanName, bool stepEngine,
                          const essentials::Identifier& agentID)
         : _ctx(ctx)
+        , _scheduler(std::max(1u, std::thread::hardware_concurrency()), this)
```

Comment:
Using a config parameter for the number of threads would be beneficial. Many people do not look too kindly on a system component that spawns hardware_concurrency many threads, when a lower number would be sufficient.